packages/neo-one-monitor/src/BrowserMetricsFactory.js

```diff
@@ -0,0 +1,144 @@
+/* @flow */
+import type {
+  MetricsFactory,
+  MetricLabels,
+  MetricConstruct,
+} from './MonitorBase';
+
+import type { Counter, Gauge, Histogram, Summary } from './types';
+
+let metrics = {
+  counters: {},
+  histograms: {},
+};
+
+export const resetMetricsForTesting = (): void => {
+  metrics = {
+    counters: {},
+    histograms: {},
+  };
+};
+
+export type MetricValue = {|
+  countOrLabels?: number | MetricLabels,
+  count?: number,
+|};
+
+export class CollectingMetric {
+  metric: MetricConstruct;
+  values: Array<MetricValue>;
+
+  constructor(metric: MetricConstruct) {
+    this.metric = metric;
+    this.values = [];
+  }
+
+  reset(): void {
+    this.values = [];
+  }
+}
+
+export type MetricCollection = {|
+  counters: { [name: string]: CollectingMetric },
+  histograms: { [name: string]: CollectingMetric },
+|};
+
+class BrowserCounter extends CollectingMetric implements Counter {
+  inc(countOrLabels?: number | MetricLabels, count?: number): void {
```

Comment:
I think this should actually always be a `MetricLabels`, can you update the definition of `Counter` (and the other metrics) to reflect that?